fsspec/tests/test_api.py

```diff
@@ -158,6 +161,57 @@ def test_chained_fs():
     assert os.listdir(d2) == ["f1"]
 
 
+@pytest.mark.xfail(reason="see issue #334", strict=True)
+def test_multilevel_chained_fs():
+    """This test reproduces intake/filesystem_spec#334"""
+    import zipfile
+
+    d1 = tempfile.mkdtemp()
+    f1 = os.path.join(d1, "f1.zip")
+    with zipfile.ZipFile(f1, mode="w") as z:
+        # filename, content
+        z.writestr("foo.txt", "foo.txt")
+        z.writestr("bar.txt", "bar.txt")
+
+    # We expected this to be the correct syntax
+    with pytest.raises(IsADirectoryError):
+        of = fsspec.open_files(f"zip://*.txt::simplecache::file://{f1}")
+        assert len(of) == 2
+
+    # But this is what is actually valid...
+    of = fsspec.open_files(f"zip://*.txt::simplecache://{f1}::file://")
+
+    assert len(of) == 2
+    for open_file in of:
+        with open_file as f:
+            assert f.read().decode("utf-8") == f.name
+
+
+def test_multilevel_chained_fs_zip_zip_file():
+    """This test reproduces intake/filesystem_spec#334"""
+    import zipfile
+
+    d1 = tempfile.mkdtemp()
+    f1 = os.path.join(d1, "f1.zip")
+    f2 = os.path.join(d1, "f2.zip")
+    with zipfile.ZipFile(f1, mode="w") as z:
+        # filename, content
+        z.writestr("foo.txt", "foo.txt")
+        z.writestr("bar.txt", "bar.txt")
+
+    with zipfile.ZipFile(f2, mode="w") as z:
+        with open(f1, "rb") as f:
+            z.writestr("f1.zip", f.read())
+
+    # We expected this to be the correct syntax
+    of = fsspec.open_files(f"zip://*.txt::zip://f1.zip::file://{f2}")
```

Comment:
This example works without any issues. This indicates to me that the issue is in the blockcache code base and not in the url parsing/unchaining.

Reply:
In this example you have some txt files in f1.zip, which in turn is within f2.zip? That is how I read that URL.

Reply:
Yes, that is correct.